### DISCUSSION TITLE
How to pass state from one element to multiple others?

### DISCUSSION THREAD

**Discussion by wgong** [zauberzeug/nicegui#1590] [Q&A]
I stumbled on nicegui yesterday while looking for streamlit alternative. The large  number of UI elements is impressive and attractive.
Thank you!!!

Interested in learning how to pass state from 1 element to multiple other elements.

My use-case is a media browser:
On left, a tree navigator displaying list of objects (e.g. animal names)
On right, 4 sections to display corresponding text, image, audio, video for a selected object.

I read  "Refreshable UI with parameters". 
Not sure if it meets my requirement as it shows 1-to-1 communication.

Is it a best practice to create a global var ourselves to pass state around?

**Comment by rodja** [comment]
Maybe a [refreshable with parameters](https://nicegui.io/documentation/refreshable#refreshable_ui_with_parameters) is what you need. Just pass the animal name to your refreshable and build the UI elements you need according to the parameter.

**Reply by wgong** [reply]
Thanks, @rodja 
will try out refreshable.

Can you also clarify on the session_state question?

**Reply by rodja** [reply]
Oh. Please do not mix two questions into one. The headline should always name the topic of one discussion. I'll change the title and remove your question "Is there global session_state dictionary like streamlit in nicegui". The answer is easily found in the docs: https://nicegui.io/documentation/storage
